source/code/plugins/auditd_plugin_lib.rb

```diff
@@ -63,4 +64,72 @@ def transform_and_wrap(event, hostname, time)
         end
 
     end # class
+
+    class AuditdDSCLog
+
+        def initialize(log)
+            @log = log
+        end
+
+        def transform(record, hostname, time)
+            operation_name = "#failed#"
+            operation_result = ""
+            dsc_version = ""
+            has_auditd = false
+            auditd_pid = 0
+            auomscollect_pid = 0
+            auoms_pid = 0
+            status_regex = /^STATUS:(?<operation_name>(Test_Start|Test_End|Set_Start|Set_End)):<(?<operation_result>[^>]*)>:(?<version>[^:]*):(?<has_auditd>(true|false)):(?<auditd_pid>[0-9]+):(?<auomscollect_pid>[0-9]+):(?<auoms_pid>[0-9]+)/
+            status_regex.match(record["message"]) { |match|
+                operation_name = match["operation_name"]
+                operation_result = match["operation_result"]
+                dsc_version = match["version"]
+                if match["has_auditd"] == "true"
+                    has_auditd = true
+                end
+                auditd_pid = match["auditd_pid"].to_i
+                auomscollect_pid = match["auomscollect_pid"].to_i
+                auoms_pid = match["auoms_pid"].to_i
+            }
+            if operation_name == "#failed#"
+                @log.error "Failed to parse STATUS message in DSC Log"
+                return {}
+            end
+            dataitem = {}
+            dataitem["Timestamp"] = OMS::Common.format_time(time)
+            dataitem["Computer"] = hostname
+            dataitem["RecordType"] = "AUOMS_METRIC"
+            dataitem["RecordTypeCode"] = 10006
+            dataitem["Namespace"] = "DSC"
+            dataitem["Name"] = operation_name
+            dataitem["version"] = dsc_version
+            dataitem["Message"] = operation_result
+            dataitem["Data"] = {
+                "has_auditd"=>has_auditd,
+                "auditd_pid"=>auditd_pid,
+                "auomscollect_pid"=>auomscollect_pid,
+                "auoms_pid"=>auoms_pid
+            }.to_json
+
+            @log.info "nxOMSAuditdPlugin STATUS: " + dataitem.to_s
```

Comment:
how often does this execute? ideally far less than 1x/min

Reply:
The DSC module runs once every 15 minutes (driven by OMSConsistencyInvoker cron job). And there are two messages per Op (Test & Set), so typically, 4 per 15-minutes.